src/main/java/uk/gov/justice/hmpps/prison/repository/sql/ScheduleRepositorySql.kt

```diff
@@ -194,13 +194,13 @@ enum class ScheduleRepositorySql(val sql: String) {
         OIS.COMMENT_TEXT AS "COMMENT",
         OIS.TO_INTERNAL_LOCATION_ID AS LOCATION_ID
         FROM OFFENDER_IND_SCHEDULES OIS
-        INNER JOIN OFFENDER_BOOKINGS OB ON OB.OFFENDER_BOOK_ID = OIS.OFFENDER_BOOK_ID AND OB.ACTIVE_FLAG = 'Y'
+        INNER JOIN OFFENDER_BOOKINGS OB ON OB.OFFENDER_BOOK_ID = OIS.OFFENDER_BOOK_ID AND OB.ACTIVE_FLAG = 'Y' AND OIS.AGY_LOC_ID = OB.AGY_LOC_ID
         INNER JOIN OFFENDERS O ON OB.OFFENDER_ID = O.OFFENDER_ID
                 INNER JOIN AGENCY_INTERNAL_LOCATIONS AIL ON OB.LIVING_UNIT_ID = AIL.INTERNAL_LOCATION_ID
                 LEFT JOIN REFERENCE_CODES RC2 ON RC2.CODE = OIS.EVENT_SUB_TYPE AND RC2.DOMAIN = 'INT_SCH_RSN'
         WHERE OIS.TO_INTERNAL_LOCATION_ID = :locationId
         AND OIS.EVENT_TYPE = 'APP'
-        AND OIS.EVENT_STATUS = 'SCH'
+        AND OIS.EVENT_STATUS <> 'CANC'
         AND OIS.EVENT_DATE >= TRUNC(COALESCE(:fromDate, OIS.EVENT_DATE))
```

Comment:
As it was, it was hiding appointments from the schedule views 1 day after the appointment, when the EVENT_STATUS is set to EXP by the Nomis `flush_schedules` job. The other issue is that in the high-security establishments, where internal movements are confirmed, these appointments would disappear from schedules and unlock lists when confirmed.